packages/sanity/src/structure/panes/document/statusBar/DocumentStatusBarActions.tsx

```diff
@@ -56,31 +59,49 @@ function DocumentStatusBarActionsInner(props: DocumentStatusBarActionsInnerProps
     )
   }, [firstActionState])
 
+  /* Version / Bundling handling */
+  const {currentVersion, isDraft} = useContext(VersionContext)
+
+  // eslint-disable-next-line no-warning-comments
+  /* TODO - replace with real data
+   - needs to check for publish data
+   - if it's published
+   */
+  const isReady = false
+
   return (
     <Flex align="center" gap={1}>
       {__internal_tasks && __internal_tasks.footerAction}
       {firstActionState && (
         <LayerProvider zOffset={200}>
           <Tooltip disabled={!tooltipContent} content={tooltipContent} placement="top">
             <Stack>
-              <Button
-                data-testid={`action-${firstActionState.label}`}
-                disabled={
-                  disabled || Boolean(firstActionState.disabled) || isTreeArrayEditingEnabledOpen
-                }
-                icon={firstActionState.icon}
-                // eslint-disable-next-line react/jsx-handler-names
-                onClick={firstActionState.onHandle}
-                ref={setButtonElement}
-                size="large"
-                text={firstActionState.label}
-                tone={firstActionState.tone || 'primary'}
-              />
+              {isDraft ? (
+                <Button
+                  data-testid={`action-${firstActionState.label}`}
+                  disabled={
+                    disabled || Boolean(firstActionState.disabled) || isTreeArrayEditingEnabledOpen
+                  }
+                  icon={firstActionState.icon}
+                  // eslint-disable-next-line react/jsx-handler-names
+                  onClick={firstActionState.onHandle}
+                  ref={setButtonElement}
+                  text={firstActionState.label}
+                  tone={firstActionState.tone || 'primary'}
+                />
+              ) : (
+                <BundleActions
+                  currentVersion={currentVersion}
+                  documentId={documentId}
+                  documentType={documentType}
+                  isReady={isReady}
+                />
```

Comment:
Thinking this should be part of the plugin once set, could we add a TODO here to review this change later?

The plugin could control which actions it allows, and it could also push the `Add to version` action as the first action. Without updating this component

Reply:
Good point, adding to this and next comment